src/System.Runtime.Serialization.Formatters/tests/SerializationTypes.cs

```diff
@@ -468,13 +517,17 @@ public static bool ArraysAreEqual(Array array1, Array array2)
 
         public static bool ArraysAreEqual<T>(T[][] array1, T[][] array2)
         {
-            if (array1 == null || array2 == null) return array1 == array2;
-            if (array1.Length != array2.Length) return false;
+            if (array1 == null || array2 == null)
+                return array1 == array2;
+            if (array1.Length != array2.Length)
+                return false;
             for (int i = 0; i < array1.Length; i++)
             {
                 T[] sub1 = array1[i], sub2 = array2[i];
-                if (sub1 == null || sub2 == null && (sub1 != sub2)) return false;
-                if (sub1.Length != sub2.Length) return false;
+                if (sub1 == null || sub2 == null && (sub1 != sub2))
```

Comment:
Please ou use parentheses when mixing || and && so I don't have to remember precedence order.

Also, if sub1 == sub2 == null then this will fail on the length check. Should it be 
``` c#
if (sub1 == null || sub2 == null)
     return sub1 == sub2;
```